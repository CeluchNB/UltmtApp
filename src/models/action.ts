import { DisplayUser } from '../types/user'
import { Realm } from '@realm/react'
import { TeamNumber } from '../types/team'
import { ActionType, Comment, LiveServerActionData } from '../types/action'

export const CommentSchema: Realm.ObjectSchema = {
    name: 'Comment',
    embedded: true,
    properties: {
        user: 'DisplayUser',
        comment: 'string',
        commentNumber: 'int',
    },
}

export class ActionSchema {
    static schema: Realm.ObjectSchema = {
        name: 'Action',
        primaryKey: '_id',
        properties: {
            _id: 'objectId',
            pointId: 'string',
            actionType: 'string',
            actionNumber: 'int',
            playerOne: 'DisplayUser?',
            playerTwo: 'DisplayUser?',
            tags: 'string[]',
            teamNumber: 'string',
            comments: 'Comment[]',
        },
    }

    _id: Realm.BSON.ObjectId
    pointId: string
    actionType: ActionType
    playerOne?: DisplayUser
    playerTwo?: DisplayUser
    tags: string[]
    actionNumber: number
    teamNumber: TeamNumber
    comments: Comment[]

    constructor(action: LiveServerActionData, pointId: string) {
        this._id = new Realm.BSON.ObjectID()
        this.pointId = pointId
        this.actionType = action.actionType
        this.playerOne = action.playerOne
        this.playerTwo = action.playerTwo
        this.tags = action.tags
        this.actionNumber = action.actionNumber
        this.teamNumber = action.teamNumber
        this.comments = action.comments
    }
}
