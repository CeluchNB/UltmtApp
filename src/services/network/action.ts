import { API_KEY, API_URL_V1 } from 'react-native-dotenv'
import axios, { AxiosResponse } from 'axios'

export const addComment = async (
    token: string,
    actionId: string,
    comment: string,
): Promise<AxiosResponse> => {
    return await axios.post(
        `${API_URL_V1}/action/${actionId}/comment`,
        { comment },
        {
            headers: {
                'X-API-Key': API_KEY,
                Authorization: `Bearer ${token}`,
            },
        },
    )
}
