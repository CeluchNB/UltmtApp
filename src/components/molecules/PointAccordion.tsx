import ActionDisplayItem from '../atoms/ActionDisplayItem'
import { GuestTeam } from '../../types/team'
import { List } from 'react-native-paper'
import Point from '../../types/point'
import { ServerAction } from '../../types/action'
import { isLivePoint } from '../../utils/point'
import { useColors } from '../../hooks'
import {
    ActivityIndicator,
    Animated,
    FlatList,
    StyleSheet,
    Text,
    View,
} from 'react-native'
import React, { useEffect } from 'react'
import { size, weight } from '../../theme/fonts'

interface PointAccordionProps {
    point: Point
    actions: ServerAction[]
    expanded: boolean
    loading: boolean
    teamOne: GuestTeam
    teamTwo: GuestTeam
    onActionPress: (action: ServerAction) => void
}

const AccordionRightView = (props: { point: Point; isExpanded: boolean }) => {
    const { colors } = useColors()
    const liveOpacity = React.useRef(new Animated.Value(0)).current
    const isLive = React.useMemo(() => {
        return isLivePoint(props.point)
    }, [props.point])

    useEffect(() => {
        if (isLive) {
            Animated.loop(
                Animated.sequence([
                    Animated.timing(liveOpacity, {
                        toValue: 1,
                        duration: 750,
                        useNativeDriver: true,
                    }),
                    Animated.timing(liveOpacity, {
                        toValue: 0,
                        duration: 750,
                        useNativeDriver: true,
                    }),
                ]),
            ).start()
        }
    }, [isLive, liveOpacity])

    const styles = StyleSheet.create({
        circleContainer: {
            marginBottom: -10,
        },
        circle: {
            width: 10,
            height: 10,
            borderRadius: 5,
            backgroundColor: 'red',
            alignSelf: 'flex-end',
        },
    })
    return (
        <View>
            {isLive && (
                <View style={styles.circleContainer}>
                    <Animated.View
                        style={[styles.circle, { opacity: liveOpacity }]}
                    />
                </View>
            )}
            <List.Icon
                {...props}
                color={colors.textPrimary}
                icon={props.isExpanded ? 'chevron-up' : 'chevron-down'}
            />
        </View>
    )
}

const getAccordionRightView = (point: Point) => {
    return (props: { isExpanded: boolean }) => {
        return (
            <AccordionRightView point={point} isExpanded={props.isExpanded} />
        )
    }
}

const PointAccordion: React.FC<PointAccordionProps> = ({
    point,
    actions,
    expanded,
    loading,
    teamOne,
    teamTwo,
    onActionPress,
}) => {
    const { colors } = useColors()

    const styles = StyleSheet.create({
        accordion: {
            backgroundColor: colors.primary,
            color: colors.textPrimary,
            borderColor: expanded ? colors.textSecondary : colors.textPrimary,
            borderStyle: 'solid',
            borderWidth: 1,
            padding: 0,
        },
        item: {
            marginBottom: 5,
            width: '90%',
            alignSelf: 'center',
        },
        titleContainer: {
            display: 'flex',
            flexDirection: 'row',
        },
        teamContainer: {
            marginRight: 10,
        },
        titleText: {
            color: colors.textPrimary,
            fontSize: size.fontFifteen,
            fontWeight: weight.bold,
        },
        circleContainer: {
            marginBottom: -10,
        },
        circle: {
            width: 10,
            height: 10,
            borderRadius: 5,
            backgroundColor: 'red',
            alignSelf: 'flex-end',
        },
    })

    return (
        <View style={{ backgroundColor: colors.primary }}>
            <List.Accordion
                id={point._id}
                style={styles.accordion}
                right={getAccordionRightView(point)}
                titleStyle={{ color: colors.textPrimary }}
                title={
                    <View style={styles.titleContainer}>
                        <View style={styles.teamContainer}>
                            <Text style={styles.titleText}>{teamOne.name}</Text>
                            <Text style={styles.titleText}>{teamTwo.name}</Text>
                        </View>
                        <View>
                            <Text style={styles.titleText}>
                                {point.teamOneScore}
                            </Text>
                            <Text style={styles.titleText}>
                                {point.teamTwoScore}
                            </Text>
                        </View>
                    </View>
                }>
                {loading && (
                    <ActivityIndicator
                        size="small"
                        color={colors.textPrimary}
                    />
                )}
                {!loading && (
                    <FlatList
                        data={actions}
                        renderItem={({ item }) => {
                            return (
                                <View
                                    key={item.actionNumber}
                                    style={styles.item}>
                                    <ActionDisplayItem
                                        action={item}
                                        onPress={onActionPress}
                                        teamOne={teamOne}
                                        teamTwo={teamTwo}
                                    />
                                </View>
                            )
                        }}
                    />
                )}
            </List.Accordion>
        </View>
    )
}

export default PointAccordion
