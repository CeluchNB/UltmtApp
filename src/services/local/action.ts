import * as Constants from '../../utils/constants'
import AsyncStorage from '@react-native-async-storage/async-storage'
import { Realm } from '@realm/react'
import { getRealm } from '../../models/realm'
import { throwApiError } from '../../utils/service-utils'
import { ActionSchema, PointSchema } from '../../models'
import { LiveServerAction, SavedServerAction } from '../../types/action'
import { getPointById, savePoint } from './point'

/**
 * Method to save actions locally
 * @param pointId id of point actions belong to
 * @param actions list of action objects
 */
export const saveActions = async (
    pointId: string,
    actions: SavedServerAction[],
) => {
    const pairs: [string, string][] = actions.map(action => [
        `${pointId}:${action._id}`,
        JSON.stringify(action),
    ])
    await AsyncStorage.multiSet(pairs)
}

/**
 * Method to get locally saved actions
 * @param pointId id of point actions belong to
 * @param actionIds list of ids of actions
 * @returns saved user ids or empty if a point is not found
 */
export const getActions = async (
    pointId: string,
    actionIds: string[],
): Promise<SavedServerAction[]> => {
    const keys = actionIds.map(id => `${pointId}:${id}`)
    const actions = await AsyncStorage.multiGet(keys)
    for (const kvp of actions) {
        if (!kvp[1]) {
            return []
        }
    }
    return actions.map(kvp => JSON.parse(kvp[1] || ''))
}

/**
 * Method to remove all locally stored actions from a given point
 * @param pointId point id to remove actions for
 */
export const deleteAllActionsByPoint = async (pointId: string) => {
    const keys = await AsyncStorage.getAllKeys()
    const filteredKeys = keys.filter(key => key.includes(pointId))
    await AsyncStorage.multiRemove(filteredKeys)
}

export const upsertAction = async (
    action: LiveServerAction,
    pointId: string,
): Promise<LiveServerAction> => {
    const realm = await getRealm()
    const point = await realm.objectForPrimaryKey<PointSchema>('Point', pointId)
    if (!point) {
        return throwApiError({}, Constants.GET_POINT_ERROR)
    }
    realm.write(() => {
        const rAction = realm.create<ActionSchema>(
            'Action',
            new ActionSchema(action, pointId),
            Realm.UpdateMode.Modified,
        )
        if (action.teamNumber === 'one') {
            point.teamOneActions = [
                ...new Set([
                    ...point.teamOneActions,
                    rAction._id.toHexString(),
                ]),
            ]
        } else {
            point.teamTwoActions = [
                ...new Set([
                    ...point.teamTwoActions,
                    rAction._id.toHexString(),
                ]),
            ]
        }
    })
    const actions = await realm.objects<ActionSchema>('Action')
    const result = actions.filtered(
        `teamNumber == "${action.teamNumber}" && actionNumber == ${action.actionNumber} && pointId == "${pointId}"`,
    )[0]

    if (!result) {
        return throwApiError({}, Constants.GET_ACTION_ERROR)
    }
    return parseAction(result)
}

export const saveMultipleServerActions = async (
    actions: LiveServerAction[],
    pointId: string,
): Promise<void> => {
    const realm = await getRealm()

    realm.write(() => {
        for (const action of actions) {
            realm.create(
                'Action',
                { ...action, _id: new Realm.BSON.ObjectId(), pointId },
                Realm.UpdateMode.All,
            )
        }
    })
}

export const deleteAction = async (
    teamNumber: 'one' | 'two',
    actionNumber: number,
    pointId: string,
): Promise<LiveServerAction> => {
    const realm = await getRealm()
    const actions = await realm.objects<ActionSchema>('Action')
    const action = actions.filtered(
        `teamNumber == "${teamNumber}" && actionNumber == ${actionNumber} && pointId == "${pointId}"`,
    )[0]

    // remove action from point array
    const point = await getPointById(pointId)
    if (teamNumber === 'one') {
        point.teamOneActions.splice(actionNumber - 1)
    } else {
        point.teamTwoActions.splice(actionNumber - 1)
    }
    await savePoint(point)

    const result = parseLiveAction(action)
    realm.write(() => {
        realm.delete(action)
    })
    return result
}

export const deleteEditableActionsByPoint = async (
    team: 'one' | 'two',
    pointId: string,
): Promise<void> => {
    const realm = await getRealm()
    const actions = await realm
        .objects<ActionSchema>('Action')
        .filtered(`teamNumber == "${team}" && pointId == "${pointId}"`)
    realm.write(() => {
        realm.delete(actions)
    })
}

export const getActionById = async (
    actionId: string,
): Promise<LiveServerAction> => {
    const realm = await getRealm()
    const action = await realm.objectForPrimaryKey<ActionSchema>(
        'Action',
        actionId,
    )
    if (!action) {
        return throwApiError({}, Constants.GET_ACTION_ERROR)
    }
    return parseLiveAction(action)
}

export const getActionsByPoint = async (
    pointId: string,
): Promise<(LiveServerAction & { _id: string; pointId: string })[]> => {
    const realm = await getRealm()
    const actions = await realm
        .objects<ActionSchema>('Action')
        .filtered(`pointId == "${pointId}"`)

    return actions.map(action => {
        return parseAction(action)
    })
}

const parseLiveAction = (schema: ActionSchema): LiveServerAction => {
    return JSON.parse(
        JSON.stringify({
            actionType: schema.actionType,
            actionNumber: schema.actionNumber,
            teamNumber: schema.teamNumber,
            tags: schema.tags,
            comments: schema.comments,
            playerOne: schema.playerOne,
            playerTwo: schema.playerTwo,
        }),
    )
}

const parseAction = (
    schema: ActionSchema,
): LiveServerAction & { _id: string; pointId: string } => {
    return JSON.parse(
        JSON.stringify({
            _id: schema._id.toHexString(),
            pointId: schema.pointId,
            actionType: schema.actionType,
            actionNumber: schema.actionNumber,
            teamNumber: schema.teamNumber,
            tags: schema.tags,
            comments: schema.comments,
            playerOne: schema.playerOne,
            playerTwo: schema.playerTwo,
        }),
    )
}
