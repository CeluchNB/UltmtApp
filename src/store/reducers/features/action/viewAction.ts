import { RootState } from '../../../store'
import { createSlice } from '@reduxjs/toolkit'
import { LiveServerAction, SavedServerAction } from '../../../../types/action'

export interface ViewActionSlice {
    liveAction: LiveServerAction | undefined
    savedAction: SavedServerAction | undefined
}

const initialState: ViewActionSlice = {
    liveAction: undefined,
    savedAction: undefined,
}

const viewActionSlice = createSlice({
    name: 'viewAction',
    initialState,
    reducers: {
        setLiveAction(state, action) {
            state.liveAction = action.payload
        },
        setSavedAction(state, action) {
            state.savedAction = action.payload
        },
    },
})

export const selectLiveAction = (state: RootState) =>
    state.viewAction.liveAction
export const selectSavedAction = (state: RootState) =>
    state.viewAction.savedAction
export const { setLiveAction, setSavedAction } = viewActionSlice.actions
export default viewActionSlice.reducer
