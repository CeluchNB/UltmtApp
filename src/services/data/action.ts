import * as Constants from '../../utils/constants'
import EncryptedStorage from 'react-native-encrypted-storage'
import { saveActions as localSaveActions } from '../local/action'
import { addComment as networkAddComment } from '../network/action'
import { throwApiError } from '../../utils/service-utils'
import { withToken } from './auth'
import {
    ClientAction,
    SavedServerAction,
    SubscriptionObject,
} from '../../types/action'
import {
    addComment as networkAddLiveComment,
    createAction as networkCreateAction,
    joinPoint as networkJoinPoint,
    nextPoint as networkNextPoint,
    subscribe as networkSubscribe,
    undoAction as networkUndoAction,
    unsubscribe as networkUnsubscribe,
} from '../network/live-action'

/**
 * Method to join a point
 * @param gameId id of game to join
 * @param pointId id of point to join
 */
export const joinPoint = async (gameId: string, pointId: string) => {
    await networkJoinPoint(gameId, pointId)
}

/**
 * Method to create an action
 * @param action action data
 * @param pointId point action belongs to
 */
export const addAction = async (action: ClientAction, pointId: string) => {
    await networkCreateAction(action, pointId)
}

/**
 * Method to undo an action
 * @param pointId undo last action on point of this id
 */
export const undoAction = async (pointId: string) => {
    await networkUndoAction(pointId)
}

/**
 * Method to add a comment to a live action
 * @param gameId id of game
 * @param pointId id of point
 * @param actionNumber action number
 * @param teamNumber 'one' | 'two' - team reporting action
 * @param comment comment text
 */
export const addLiveComment = async (
    gameId: string,
    pointId: string,
    actionNumber: number,
    teamNumber: 'one' | 'two',
    comment: string,
) => {
    const token = (await EncryptedStorage.getItem('access_token')) || ''
    await networkAddLiveComment(
        token,
        gameId,
        pointId,
        actionNumber,
        teamNumber,
        comment,
    )
}

/**
 * Method to indicate the next point has been started
 * @param pointId id of point that was just finished
 */
export const nextPoint = async (pointId: string) => {
    await networkNextPoint(pointId)
}

/**
 * Method to listen to specific action events
 * @param subscriptions object containing subscription types and methods
 */
export const subscribe = async (subscriptions: SubscriptionObject) => {
    await networkSubscribe(subscriptions)
}

/**
 * Method to unsubscribe from all action websckets
 */
export const unsubscribe = () => {
    networkUnsubscribe()
}

/**
 * Method to add a comment to a saved action.
 * @param actionId id of action
 * @param pointId id of point
 * @param comment comment text
 * @returns updated action
 */
export const addComment = async (
    actionId: string,
    pointId: string,
    comment: string,
): Promise<SavedServerAction> => {
    try {
        const response = await withToken(networkAddComment, actionId, comment)
        const { action } = response.data
        await localSaveActions(pointId, [action])
        return action
    } catch (error) {
        return throwApiError(error, Constants.COMMENT_ERROR)
    }
}
