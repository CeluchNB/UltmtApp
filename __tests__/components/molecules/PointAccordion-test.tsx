import { List } from 'react-native-paper'
import Point from '../../../src/types/point'
import PointAccordion from '../../../src/components/molecules/PointAccordion'
import React from 'react'
import { render } from '@testing-library/react-native'
import { Action, ActionFactory, ActionType } from '../../../src/types/action'

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')
jest.mock('react-native-google-mobile-ads', () => {
    return {
        default: { initialize: jest.fn(), setRequestConfiguration: jest.fn() },
        MaxAdsContentRating: { T: 'T', PG: 'PG' },
        BannerAd: 'Ad',
        BannerAdSize: { BANNER: 'banner' },
        TestIds: { BANNER: 'bannertest' },
    }
})

const point: Point = {
    _id: 'point1',
    pointNumber: 1,
    pullingTeam: { name: 'Temper' },
    receivingTeam: { name: 'Truck' },
    teamOnePlayers: [],
    teamTwoPlayers: [],
    teamOneScore: 2,
    teamTwoScore: 1,
    teamOneActive: true,
    teamTwoActive: false,
    teamOneActions: [],
    teamTwoActions: [],
}

const actions: Action[] = [
    {
        comments: [],
        tags: ['huck'],
        actionNumber: 1,
        actionType: ActionType.CATCH,
        teamNumber: 'one',
        playerOne: {
            _id: 'user1',
            firstName: 'First 1',
            lastName: 'Last 1',
            username: 'firstlast1',
        },
        playerTwo: {
            _id: 'user2',
            firstName: 'First 2',
            lastName: 'Last 2',
            username: 'firstlast2',
        },
    },
    {
        comments: [],
        tags: [],
        actionNumber: 2,
        actionType: ActionType.CATCH,
        teamNumber: 'one',
        playerTwo: {
            _id: 'user1',
            firstName: 'First 1',
            lastName: 'Last 1',
            username: 'firstlast1',
        },
        playerOne: {
            _id: 'user2',
            firstName: 'First 2',
            lastName: 'Last 2',
            username: 'firstlast2',
        },
    },
].map(a => ActionFactory.createFromAction(a))

const onActionPress = jest.fn()

describe('PointAccordion', () => {
    it('should match snapshot closed', () => {
        const snapshot = render(
            <List.AccordionGroup expandedId="other1">
                <PointAccordion
                    point={point}
                    actions={actions}
                    loading={false}
                    expanded={false}
                    teamOne={{ name: 'Temper' }}
                    teamTwo={{ name: 'Truck' }}
                    error={''}
                    onActionPress={onActionPress}
                />
            </List.AccordionGroup>,
        )

        expect(snapshot.getByText('Temper')).toBeTruthy()
        expect(snapshot.getByText('Truck')).toBeTruthy()
        expect(snapshot.getByText('1')).toBeTruthy()
        expect(snapshot.getByText('2')).toBeTruthy()

        expect(snapshot).toMatchSnapshot()
    })

    it('should match snapshot open', async () => {
        const snapshot = render(
            <List.AccordionGroup expandedId={point._id}>
                <PointAccordion
                    point={point}
                    actions={actions}
                    loading={false}
                    expanded={true}
                    teamOne={{ name: 'Temper' }}
                    teamTwo={{ name: 'Truck' }}
                    error={''}
                    onActionPress={onActionPress}
                />
            </List.AccordionGroup>,
        )

        expect(snapshot).toMatchSnapshot()
    })

    it('displays error', () => {
        const { getByText } = render(
            <List.AccordionGroup expandedId={point._id}>
                <PointAccordion
                    point={point}
                    actions={actions}
                    loading={false}
                    expanded={false}
                    teamOne={{ name: 'Temper' }}
                    teamTwo={{ name: 'Truck' }}
                    error={'test error'}
                    onActionPress={onActionPress}
                />
            </List.AccordionGroup>,
        )

        expect(getByText('test error')).toBeTruthy()
    })
})
