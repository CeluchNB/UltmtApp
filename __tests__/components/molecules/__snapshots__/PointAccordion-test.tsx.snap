// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PointAccordion should match snapshot closed 1`] = `
<View
  style={
    {
      "backgroundColor": "#303030",
    }
  }
>
  <View>
    <View
      style={
        {
          "backgroundColor": "#303030",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "disabled": false,
            "expanded": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            false,
            [
              {
                "padding": 8,
              },
              {
                "backgroundColor": "#303030",
                "borderColor": "#ffd600",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 1,
                "color": "#ffd600",
                "margin": 4,
                "padding": 0,
              },
            ],
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              [
                {
                  "margin": 8,
                },
                {
                  "flex": 1,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontSize": 16,
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    {
                      "color": "#ffd600",
                    },
                  ],
                ]
              }
            >
              <View
                style={
                  {
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    {
                      "marginRight": 10,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#ffd600",
                        "fontSize": 30.821917808219176,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Temper
                  </Text>
                  <Text
                    style={
                      {
                        "color": "#ffd600",
                        "fontSize": 30.821917808219176,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Truck
                  </Text>
                </View>
                <View>
                  <Text
                    style={
                      {
                        "color": "#ffd600",
                        "fontSize": 30.821917808219176,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    2
                  </Text>
                  <Text
                    style={
                      {
                        "color": "#ffd600",
                        "fontSize": 30.821917808219176,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    1
                  </Text>
                </View>
              </View>
            </Text>
          </View>
          <View
            style={
              [
                {
                  "margin": 8,
                },
                undefined,
              ]
            }
          >
            <View>
              <View
                style={
                  {
                    "marginBottom": -10,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "alignSelf": "flex-end",
                      "backgroundColor": "red",
                      "borderRadius": 5,
                      "height": 10,
                      "opacity": 0,
                      "width": 10,
                    }
                  }
                />
              </View>
              <View
                pointerEvents="box-none"
                style={
                  [
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#ffd600",
                        "fontSize": 24,
                      },
                      [
                        {
                          "lineHeight": 24,
                          "transform": [
                            {
                              "scaleX": 1,
                            },
                          ],
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                      ],
                      {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  ó°…€
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`PointAccordion should match snapshot open 1`] = `
<View
  style={
    {
      "backgroundColor": "#303030",
    }
  }
>
  <View>
    <View
      style={
        {
          "backgroundColor": "#303030",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "disabled": false,
            "expanded": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            false,
            [
              {
                "padding": 8,
              },
              {
                "backgroundColor": "#303030",
                "borderColor": "#31adff",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 1,
                "color": "#ffd600",
                "margin": 4,
                "padding": 0,
              },
            ],
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              [
                {
                  "margin": 8,
                },
                {
                  "flex": 1,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontSize": 16,
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    {
                      "color": "#ffd600",
                    },
                  ],
                ]
              }
            >
              <View
                style={
                  {
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    {
                      "marginRight": 10,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#ffd600",
                        "fontSize": 30.821917808219176,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Temper
                  </Text>
                  <Text
                    style={
                      {
                        "color": "#ffd600",
                        "fontSize": 30.821917808219176,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Truck
                  </Text>
                </View>
                <View>
                  <Text
                    style={
                      {
                        "color": "#ffd600",
                        "fontSize": 30.821917808219176,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    2
                  </Text>
                  <Text
                    style={
                      {
                        "color": "#ffd600",
                        "fontSize": 30.821917808219176,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    1
                  </Text>
                </View>
              </View>
            </Text>
          </View>
          <View
            style={
              [
                {
                  "margin": 8,
                },
                undefined,
              ]
            }
          >
            <View>
              <View
                style={
                  {
                    "marginBottom": -10,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "alignSelf": "flex-end",
                      "backgroundColor": "red",
                      "borderRadius": 5,
                      "height": 10,
                      "opacity": 0,
                      "width": 10,
                    }
                  }
                />
              </View>
              <View
                pointerEvents="box-none"
                style={
                  [
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#ffd600",
                        "fontSize": 24,
                      },
                      [
                        {
                          "lineHeight": 24,
                          "transform": [
                            {
                              "scaleX": 1,
                            },
                          ],
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                      ],
                      {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  ó°…€
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
